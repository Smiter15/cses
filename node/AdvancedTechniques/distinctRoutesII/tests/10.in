500 1000 30
1 180
180 108
108 22
22 233
233 100
1 176
176 425
425 179
179 255
255 191
191 254
254 477
477 207
207 100
1 32
32 373
373 43
43 246
246 198
198 127
127 21
21 374
374 137
137 100
1 357
357 50
50 158
158 275
275 379
379 462
462 433
433 59
59 70
70 455
455 100
1 53
53 408
408 494
494 231
231 440
440 185
185 484
484 195
195 170
170 19
19 100
1 43
43 61
61 249
249 100
1 465
465 47
47 114
114 102
102 104
104 248
248 151
151 100
1 301
301 31
31 279
279 234
234 89
89 348
348 466
466 158
158 178
178 152
152 100
1 409
409 369
369 114
114 171
171 341
341 100
1 467
467 345
345 325
325 305
305 100
1 343
343 270
270 418
418 234
234 372
372 100
1 482
482 23
23 61
61 283
283 330
330 100
1 294
294 197
197 465
465 39
39 354
354 100
1 283
283 100
1 337
337 396
396 1
1 455
455 180
180 467
467 300
300 100
1 272
272 172
172 347
347 320
320 90
90 100
1 192
192 388
388 341
341 491
491 449
449 401
401 322
322 100
1 472
472 19
19 308
308 11
11 100
1 451
451 100
1 432
432 280
280 93
93 328
328 58
58 48
48 8
8 302
302 126
126 290
290 100
1 75
75 414
414 172
172 164
164 273
273 141
141 52
52 100
1 132
132 278
278 292
292 453
453 344
344 265
265 249
249 152
152 53
53 100
1 380
380 308
308 330
330 438
438 178
178 159
159 100
1 4
4 100
1 76
76 406
406 233
233 427
427 480
480 148
148 376
376 144
144 420
420 17
17 100
1 312
312 148
148 29
29 381
381 379
379 150
150 146
146 128
128 79
79 476
476 100
1 459
459 100
1 134
134 175
175 18
18 292
292 447
447 21
21 236
236 300
1 246
246 227
227 184
184 393
393 380
380 105
105 92
92 396
396 77
77 181
181 100
1 382
382 62
62 201
201 32
32 485
485 106
106 100
100 239
239 132
132 347
347 300
300 44
44 22
22 317
317 500
100 246
246 115
115 348
348 47
47 318
318 95
95 51
51 2
2 265
265 500
100 106
106 500
100 105
105 460
460 314
314 426
426 500
100 154
154 404
404 374
374 500
100 11
11 262
262 155
155 419
419 109
109 500
100 462
462 407
407 327
327 298
298 500
100 441
441 423
423 500
100 38
38 295
295 250
250 39
39 500
100 458
458 422
422 194
194 63
63 160
160 286
286 500
100 2
2 217
217 393
393 153
153 133
133 403
403 414
414 65
65 323
323 500
100 74
74 63
63 207
207 400
400 360
360 120
120 62
62 336
336 500
100 356
356 363
363 250
250 416
416 321
321 500
100 388
388 384
384 109
109 451
451 151
151 111
111 168
168 321
100 78
78 3
3 455
455 419
419 500
100 255
255 492
492 165
165 240
240 171
171 500
100 68
68 364
364 500
100 220
220 77
77 500
100 414
414 397
397 478
478 80
80 59
59 87
87 500
100 486
486 197
197 253
253 86
86 238
238 500
100 88
88 470
470 27
27 467
467 3
3 296
296 132
132 242
242 500
100 434
434 119
119 311
311 76
76 332
332 90
90 74
74 500
100 396
396 500
100 83
83 374
374 344
344 419
419 460
460 152
152 183
183 158
158 500
100 268
268 500
100 12
12 500
100 420
420 38
38 500
100 200
200 111
111 10
10 442
442 355
355 443
443 338
338 443
443 297
297 500
100 311
311 370
370 355
355 207
207 500
100 216
216 81
81 500
479 239
227 252
305 233
182 96
105 271
219 139
390 368
133 292
50 470
241 416
114 39
17 20
255 381
125 257
2 191
50 332
429 276
83 86
361 116
33 465
238 251
456 127
470 440
271 19
409 363
287 374
401 303
393 155
36 370
411 389
412 460
221 341
88 155
278 448
271 162
412 8
412 367
487 382
306 257
252 66
119 38
440 20
193 184
174 228
405 437
468 317
396 188
9 335
195 138
282 465
299 46
324 62
264 310
295 422
66 47
339 185
436 130
56 128
314 229
207 70
165 27
386 413
214 246
247 260
383 381
224 181
278 400
243 393
209 37
166 127
435 5
163 371
134 218
350 299
298 409
369 463
435 106
227 500
352 325
260 234
57 335
267 186
234 9
79 295
397 244
421 184
248 435
54 234
152 255
32 449
163 252
263 449
358 341
449 61
166 208
294 74
394 60
260 128
420 338
274 169
433 194
352 33
128 257
118 131
363 1
79 378
253 194
326 462
34 126
22 51
185 167
125 79
79 236
206 498
425 479
18 357
24 221
241 3
477 358
133 192
359 63
69 463
256 246
424 142
372 297
44 56
463 20
134 393
243 179
21 260
388 44
481 480
398 309
338 30
352 48
445 420
10 52
166 433
193 37
81 237
444 395
108 78
288 363
268 30
393 289
141 132
184 121
112 82
282 301
463 133
348 407
405 357
459 422
141 3
458 221
91 253
467 199
330 254
413 450
135 305
90 276
289 273
248 400
206 381
52 169
14 399
427 270
255 385
191 247
388 500
467 330
252 433
28 82
39 440
31 173
97 472
300 385
244 48
136 302
428 187
470 293
85 248
62 191
133 252
437 372
103 255
201 355
39 81
288 77
372 170
102 468
141 401
204 384
448 339
185 228
25 6
20 461
254 434
151 238
185 439
461 288
45 13
494 84
93 281
12 465
450 465
284 90
366 488
325 165
326 362
244 203
219 264
163 472
197 166
61 233
456 21
372 1
34 365
436 478
497 299
442 446
264 78
387 481
65 64
497 242
425 241
444 143
4 459
467 52
476 27
136 283
400 8
283 285
224 70
262 73
369 56
370 484
133 257
464 49
320 460
290 96
200 86
90 55
396 56
458 371
435 94
153 186
101 288
470 176
209 231
248 429
138 118
412 270
226 227
318 397
187 460
492 238
45 433
145 440
489 102
162 275
195 166
312 147
305 281
175 14
363 274
442 1
243 354
122 468
432 292
364 470
103 207
208 147
140 204
438 480
305 451
106 352
116 417
350 421
49 24
434 263
298 227
115 40
80 237
360 364
380 75
333 482
282 392
480 273
95 417
104 252
219 61
103 334
329 452
254 229
328 39
491 125
266 106
16 197
194 375
60 425
450 393
406 83
136 237
207 231
5 310
334 223
222 436
56 50
239 162
278 66
52 120
42 317
225 58
366 270
284 277
194 233
21 451
315 157
39 21
387 43
182 220
265 255
7 173
304 97
186 433
15 237
53 56
406 129
465 271
399 249
47 444
481 68
247 148
224 285
20 462
328 54
33 444
308 391
116 464
487 153
390 448
409 295
429 373
417 179
473 315
474 454
234 220
453 309
357 472
270 36
377 154
479 185
44 447
148 383
99 43
235 340
343 143
486 123
368 402
301 340
217 274
145 302
346 97
111 54
421 232
89 297
386 419
333 281
217 480
15 316
375 250
7 69
244 493
191 111
246 343
303 314
468 447
116 165
396 226
218 168
457 306
464 194
77 149
327 293
480 341
108 354
442 467
274 186
459 316
148 204
10 450
18 478
249 133
142 144
210 212
311 18
369 126
212 445
274 390
238 106
230 197
311 24
163 85
61 473
252 208
177 262
10 46
239 258
30 232
253 91
295 415
108 164
40 319
108 166
60 197
123 142
394 433
165 408
17 225
381 269
284 409
382 293
306 472
402 187
55 6
277 350
420 384
365 312
55 324
329 114
255 266
383 419
173 400
495 405
20 279
313 401
423 470
224 177
156 278
182 432
127 102
168 491
265 222
167 93
43 442
452 425
212 476
176 59
381 47
337 45
299 111
367 22
287 22
300 321
306 278
274 473
121 38
194 287
130 380
325 24
173 128
448 385
103 476
295 335
22 131
232 321
93 98
194 232
471 345
52 276
123 325
248 95
362 293
381 343
173 205
366 197
184 165
433 286
140 227
473 14
209 204
186 302
153 379
33 123
224 436
399 198
260 146
292 473
439 24
167 463
80 32
159 263
48 444
400 188
170 372
53 379
427 238
180 79
468 64
202 43
499 452
240 110
449 383
82 387
406 100
201 337
131 212
99 30
155 351
69 176
222 121
54 1
358 85
79 178
500 132
220 498
83 460
459 32
194 40
270 452
491 323
288 473
34 239
3 40
89 71
215 162
192 121
162 401
205 241
78 205
224 150
54 307
461 13
338 154
404 459
105 395
281 245
367 166
483 369
205 423
292 272
84 483
392 98
235 448
190 165
4 413
314 58
219 274
422 408
279 325
367 384
71 499
128 438
165 462
158 221
236 449
492 319
283 235
268 370
183 457
34 38
370 199
95 440
324 16
348 102
193 66
337 263
416 464
52 80
277 62
153 12
10 496
183 145
231 450
14 265
407 399
302 128
449 249
67 272
116 414
373 160
331 210
423 247
25 326
178 154
387 330
17 249
178 199
393 408
1 258
24 259
156 177
386 456
277 452
227 393
218 451
52 400
160 326
146 37
152 176
42 390
357 58
138 34
109 382
293 109
139 316
367 146
345 104
101 121
55 179
13 124
