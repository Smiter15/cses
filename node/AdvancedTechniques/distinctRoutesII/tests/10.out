138
6
1 180 108 22 317 500 
5
1 176 59 87 500 
6
1 32 373 160 286 500 
5
1 357 50 158 500 
5
1 53 100 106 500 
6
1 43 442 355 207 500 
4
1 465 39 500 
7
1 301 31 173 205 423 500 
6
1 409 369 114 171 500 
6
1 467 300 100 38 500 
8
1 343 270 418 234 372 297 500 
4
1 294 74 500 
6
1 283 100 68 364 500 
5
1 337 396 77 500 
5
1 455 419 109 500 
7
1 272 172 347 300 321 500 
4
1 192 388 500 
6
1 472 19 100 396 500 
5
1 451 151 238 500 
8
1 432 280 93 328 39 81 500 
6
1 75 414 65 323 500 
4
1 132 242 500 
7
1 380 105 460 314 426 500 
5
1 4 100 268 500 
6
1 76 406 83 374 500 
6
1 312 148 383 419 500 
5
1 459 100 12 500 
6
1 134 175 14 265 500 
4
1 246 227 500 
5
1 382 62 336 500 
